In [43]:
import pandas as pd
import numpy as np

### 1 задача

Скачать файл в уроке. 
Считать данные с помощью pandas. 
**Вывести на экран первые 5 строк.
Посмотреть на описание признаков** и на их содержание.

In [42]:
df = pd.read_csv('./kc-house-data.csv', encoding="windows - 1251")
df.head()

id             date     price  bedrooms  bathrooms  sqft_living   
0  7129300520  20141013T000000  221900.0         3       1.00         1180  \
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement   
0      5650     1.0           0     0  ...      7        1180              0  \
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15   
0      1955             0    98178  47.5112 -122.257           1340  \
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [44]:
df.shape

(21613, 21)

In [45]:
df.dtypes

id                 int64
date              object
price            float64
bedrooms           int64
bathrooms        float64
sqft_living        int64
sqft_lot           int64
floors           float64
waterfront         int64
view               int64
condition          int64
grade              int64
sqft_above         int64
sqft_basement      int64
yr_built           int64
yr_renovated       int64
zipcode            int64
lat              float64
long             float64
sqft_living15      int64
sqft_lot15         int64
dtype: object

### 2 задача

Проведите первичный анализ данных.
Изучите типы данных.
**Найдите количество пропущенных ячеек в данных.
Посчитайте основные статистики по всем признакам** и поизучайте их.
Пишите выводы.

In [46]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [47]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living   
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000  \
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition   
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000  \
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated   
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000  \
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

**Выводы:**
    
- Пропущенных значений в таблице нет (все ячейки заполнены).
- Все поля таблицы - либо целочисленные значения (15 столбцов), либо - вещественные (5). И только столбец "date" содержит строковые значения.
- В таблице приведены данные по 20 параметрам (+id) 21613 домов.
- Средняя цена всех домов - 540 088,1, минимальная цена - 75 000, максимальная - 7 700 000 (медианная цена - 450 000).
- Медианное кол-во спален - 3.
- Средняя площадь - 2 079,899736 (жилая), - 15 106,97 (общая).
- Среднее состояние домов - 3,41, средний грейд - 7, 65.
- Самый "древний" дом - 1900 г. (постройка), самый "молодой" - 2015г.

### 3 задача

Ответьте на несколько вопросов

#### 3.1 В каком диапазоне изменяются стоимости недвижимости?

_Максимальная цена домов:_

In [49]:
df[df['price']==df['price'].max()]

id             date      price  bedrooms  bathrooms   
7252  6762700020  20141013T000000  7700000.0         6        8.0  \

      sqft_living  sqft_lot  floors  waterfront  view  ...  grade  sqft_above   
7252        12050     27600     2.5           0     3  ...     13        8570  \

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long   
7252           3480      1910          1987    98102  47.6298 -122.323  \

      sqft_living15  sqft_lot15  
7252           3940        8800  

[1 rows x 21 columns]

_Минимальная цена домов:_

In [50]:
df[df['price']==df['price'].min()]

id             date    price  bedrooms  bathrooms  sqft_living   
1149  3421079032  20150217T000000  75000.0         1        0.0          670  \

      sqft_lot  floors  waterfront  view  ...  grade  sqft_above   
1149     43377     1.0           0     0  ...      3         670  \

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long   
1149              0      1966             0    98022  47.2638 -121.906  \

      sqft_living15  sqft_lot15  
1149           1160       42882  

[1 rows x 21 columns]

_Диапазон изменяются стоимости недвижимости:_

In [48]:
df['price'].min(),df['price'].max()

(75000.0, 7700000.0)

#### 3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?

In [51]:
df['sqft_living'].mean()/df['sqft_lot'].mean()

0.1376781757959227

In [52]:
f"Жилая площадь занимает {round(df['sqft_living'].mean()/df['sqft_lot'].mean()*100, 2)}% от всей площади по всем домам."

'Жилая площадь занимает 13.77% от всей площади по всем домам.'

#### 3.3 Как много домов с разными этажами в данных?

In [53]:
df['floors'].value_counts()

floors
1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: count, dtype: int64

#### 3.4 Насколько хорошие состояния у домов в данных?

In [54]:
df['condition'].value_counts()

condition
3    14031
4     5679
5     1701
2      172
1       30
Name: count, dtype: int64

#### 3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [55]:
f"Первые дома построили в {df['yr_built'].min()} году."

'Первые дома построили в 1900 году.'

In [56]:
df[df['yr_built']==df['yr_built'].min()]

id             date     price  bedrooms  bathrooms   
14     1175000570  20150312T000000  530000.0         5       2.00  \
115    3626039325  20141121T000000  740500.0         3       3.50   
498    9274202270  20140818T000000  625000.0         2       1.50   
537    5694500105  20141204T000000  595000.0         2       2.00   
703    7011200260  20141219T000000  485000.0         4       2.00   
...           ...              ...       ...       ...        ...   
19063  1702901340  20140613T000000  718500.0         3       2.00   
19137  3388110230  20140729T000000  179000.0         4       1.75   
19319  4083302225  20141014T000000  850000.0         4       3.00   
19385  2420069042  20150424T000000  240000.0         3       2.00   
19598  7518507685  20150223T000000  400000.0         3       1.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
14            1810      4850     1.5           0     0  ...      7  \
115           4380      6350     2.0           0     0  ...      8   
498           1490      5750     1.5           0     0  ...      7   
537           1510      4000     1.0           0     0  ...      7   
703           1400      3600     1.0           0     0  ...      7   
...            ...       ...     ...         ...   ...  ...    ...   
19063         2910      6600     2.0           0     0  ...      7   
19137         1790      7175     1.5           0     0  ...      6   
19319         2550      3784     1.5           0     0  ...      8   
19385         1553      6550     1.0           0     0  ...      7   
19598         1100      5100     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
14           1810              0      1900             0    98107  47.6700  \
115          2780           1600      1900          1999    98117  47.6981   
498          1190            300      1900             0    98116  47.5872   
537          1010            500      1900             0    98103  47.6582   
703          1100            300      1900             0    98119  47.6385   
...           ...            ...       ...           ...      ...      ...   
19063        1920            990      1900          1988    98118  47.5576   
19137        1410            380      1900             0    98168  47.4963   
19319        1750            800      1900             0    98103  47.6559   
19385        1553              0      1900          2001    98022  47.2056   
19598        1100              0      1900             0    98117  47.6790   

          long  sqft_living15  sqft_lot15  
14    -122.394           1360        4850  
115   -122.368           1830        6350  
498   -122.390           1590        4025  
537   -122.345           1920        4000  
703   -122.370           1630        2048  
...        ...            ...         ...  
19063 -122.281           1370        5500  
19137 -122.318           1790        8417  
19319 -122.338           2100        4560  
19385 -121.994           1010       10546  
19598 -122.386           1540        5100  

[87 rows x 21 columns]

In [57]:
f"Последние дома построили в {df['yr_built'].max()} году."

'Последние дома построили в 2015 году.'

In [58]:
df[df['yr_built']==df['yr_built'].max()]

id             date      price  bedrooms  bathrooms   
643    9385200045  20150512T000000   729500.0         3       2.50  \
1763   1832100030  20140625T000000   597326.0         4       4.00   
2687   3076500830  20141029T000000   385195.0         1       1.00   
4154   8077100031  20150422T000000   631000.0         3       2.25   
7526   9520900210  20141231T000000   614285.0         5       2.75   
8039   1250200495  20140624T000000   455000.0         2       1.50   
8425    558100090  20150312T000000   628000.0         5       2.75   
14215  8156600210  20150326T000000  1285000.0         5       3.50   
14489  2770601530  20140826T000000   500000.0         2       2.25   
14925  2770602360  20150421T000000   671000.0         4       2.75   
17098  9126100346  20140617T000000   350000.0         3       2.00   
19722  9161100075  20150318T000000   673000.0         4       2.25   
19805  9126100765  20140801T000000   455000.0         3       1.75   
19863  2768100510  20150402T000000   649000.0         3       2.00   
19907  7853440140  20150409T000000   802945.0         5       3.50   
19945  8011100047  20150306T000000   530000.0         4       2.75   
19959  7853430690  20150127T000000   572800.0         3       2.50   
20086  2768100512  20150422T000000   659000.0         2       2.50   
20251  7853440050  20150505T000000   771005.0         5       4.50   
20311  7132300525  20150411T000000   500000.0         3       1.75   
20369  4385700250  20150407T000000  1800000.0         4       3.50   
20487  9103000455  20150424T000000   920000.0         4       3.25   
20488  8691440220  20150202T000000  1289990.0         5       4.00   
20698   255470030  20150429T000000   619990.0         4       2.75   
20770  9310300160  20140828T000000   357000.0         5       2.50   
20798   952005863  20150505T000000   643950.0         3       2.25   
20852  1257201420  20140709T000000   595000.0         4       3.25   
20881  3052700385  20150414T000000   765000.0         4       2.25   
20933  2652501565  20150423T000000  1550000.0         3       3.25   
20963  6058600220  20140731T000000   230000.0         3       1.50   
21262  5694500840  20141125T000000   559000.0         2       3.00   
21270  1773100541  20150417T000000   389950.0         3       2.25   
21316  6600000050  20150310T000000  1698000.0         4       3.50   
21333  2424059174  20150508T000000  1999950.0         4       3.25   
21369   923059259  20150401T000000   455950.0         4       2.50   
21372  6169901185  20140520T000000   490000.0         5       3.50   
21432  7104100110  20150511T000000   899000.0         4       3.50   
21501  4233600150  20150203T000000  1150000.0         5       4.25   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
643           1660      1091     3.0           0     1  ...      9  \
1763          3570      8250     2.0           0     0  ...     10   
2687           710      6000     1.5           0     0  ...      6   
4154          1670      1396     2.0           0     0  ...      9   
7526          2730      6401     2.0           0     0  ...      8   
8039          1200      1259     2.0           0     0  ...      8   
8425          2600      8160     2.0           0     0  ...      8   
14215         2980      5100     2.0           0     0  ...     10   
14489         1570      1269     2.0           0     0  ...      9   
14925         1890      1475     2.0           0     0  ...      9   
17098         1380      3600     3.0           0     0  ...      8   
19722         2580      2875     2.0           0     0  ...      9   
19805         1320      1014     3.0           0     0  ...      9   
19863         1530      1442     3.0           0     0  ...      9   
19907         4000      9234     2.0           0     0  ...      9   
19945         2740      7872     2.0           0     0  ...     10   
19959         3310      4682     2.0           0     0  ...      9   
20086         1450      

### 4 задача

Ответьте на несколько вопросов

#### 4.1 Сколько в среднем стоят дома, у которых 2 спальни?

In [59]:
f"Дома, у которых 2 спальни в среднем стоят {round(df[df['bedrooms'] == 2]['price'].mean(), 2)}"

'Дома, у которых 2 спальни в среднем стоят 401372.68'

#### 4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?

In [60]:
f"Дома, у которых стоимость больше 600 000, в среднем общая площадь равна {round(df[df['price'] > 600000]['sqft_lot'].mean(), 2)} кв.метров"

'Дома, у которых стоимость больше 600 000, в среднем общая площадь равна 20442.52 кв.метров'

#### 4.3 Как много домов коснулся ремонт?

In [61]:
f"Дома, которых коснулся ремонт - {df[df['yr_renovated'] != 0].shape[0]} штук."

'Дома, которых коснулся ремонт - 914 штук.'

#### 4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [62]:
f"Средняя стоимость домов с оценкой grade выше 10 - {round(df[df['grade'] > 10]['price'].mean(), 2)}."

'Средняя стоимость домов с оценкой grade выше 10 - 1678635.12.'

In [63]:
f"Средняя стоимость домов с оценкой grade меньше 4 - {round(df[df['grade'] < 4]['price'].mean(), 2)}."

'Средняя стоимость домов с оценкой grade меньше 4 - 189750.0.'

In [64]:
f"Разница средних цен домов с оценкой grade выше 10 и с оценкой grade меньше 4 = {round(df[df['grade'] > 10]['price'].mean() - df[df['grade'] < 4]['price'].mean(), 2)}"

'Разница средних цен домов с оценкой grade выше 10 и с оценкой grade меньше 4 = 1488885.12'

### 5 задача

Ответьте на несколько вопросов

#### 5.1 Выберите дом клиенту

Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?

_Дома с как минимум с тремя ванными:_

In [65]:
df[df['bathrooms'] >= 3]

id             date      price  bedrooms  bathrooms   
3      2487200875  20141209T000000   604000.0         4       3.00  \
5      7237550310  20140512T000000  1225000.0         4       4.50   
15     9297300055  20150124T000000   650000.0         4       3.00   
58     7922800400  20140827T000000   951000.0         5       3.25   
59     8079040320  20150223T000000   430000.0         4       3.00   
...           ...              ...        ...       ...        ...   
21593  8672200110  20150317T000000  1088000.0         5       3.75   
21597   191100405  20150421T000000  1575000.0         4       3.25   
21599  7202300110  20140915T000000   810000.0         4       3.00   
21600   249000205  20141015T000000  1537000.0         5       3.75   
21606  7936000429  20150326T000000  1007500.0         4       3.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
3             1960      5000     1.0           0     0  ...      7  \
5             5420    101930     1.0           0     0  ...     11   
15            2950      5000     2.0           0     3  ...      9   
58            3250     14342     2.0           0     4  ...      8   
59            1850      9976     2.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21593         4170      8142     2.0           0     2  ...     10   
21597         3410     10125     2.0           0     0  ...     10   
21599         3990      7838     2.0           0     0  ...      9   
21600         4470      8088     2.0           0     0  ...     11   
21606         3510      7200     2.0           0     0  ...      9   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
3            1050            910      1965             0    98136  47.5208  \
5            3890           1530      2001             0    98053  47.6561   
15           1980            970      1979             0    98126  47.5714   
58           3250              0      1968             0    98008  47.5880   
59           1850              0      1991             0    98059  47.5059   
...           ...            ...       ...           ...      ...      ...   
21593        4170              0      2006             0    98056  47.5354   
21597        3410              0      2007             0    98040  47.5653   
21599        3990              0      2003             0    98053  47.6857   
21600        4470              0      2008             0    98004  47.6321   
21606        2600            910      2009             0    98136  47.5537   

          long  sqft_living15  sqft_lot15  
3     -122.393           1360        5000  
5     -122.005           4760      101930  
15    -122.375           2140        4000  
58    -122.116           2960       11044  
59    -122.149           2270        8542  
...        ...            ...         ...  
21593 -122.181           3030        7980  
21597 -122.223           2290       10125  
21599 -122.046           3370        6814  
21600 -122.200           2780        8964  
21606 -122.398           2050        6200  

[2630 rows x 21 columns]

_Дома с подвалами:_

In [66]:
df[df['sqft_basement'] > 0]

id             date      price  bedrooms  bathrooms   
1      6414100192  20141209T000000   538000.0         3       2.25  \
3      2487200875  20141209T000000   604000.0         4       3.00   
5      7237550310  20140512T000000  1225000.0         4       4.50   
8      2414600126  20150415T000000   229500.0         3       1.00   
10     1736800520  20150403T000000   662500.0         3       2.50   
...           ...              ...        ...       ...        ...   
21588  2767600688  20141113T000000   414500.0         2       1.50   
21590  7430200100  20140514T000000  1222500.0         4       3.50   
21595  1972201967  20141031T000000   520000.0         2       2.25   
21606  7936000429  20150326T000000  1007500.0         4       3.50   
21607  2997800021  20150219T000000   475000.0         3       2.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
1             2570      7242     2.0           0     0  ...      7  \
3             1960      5000     1.0           0     0  ...      7   
5             5420    101930     1.0           0     0  ...     11   
8             1780      7470     1.0           0     0  ...      7   
10            3560      9796     1.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21588         1210      1278     2.0           0     0  ...      8   
21590         4910      9444     1.5           0     0  ...     11   
21595         1530       981     3.0           0     0  ...      8   
21606         3510      7200     2.0           0     0  ...      9   
21607         1310      1294     2.0           0     0  ...      8   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
1            2170            400      1951          1991    98125  47.7210  \
3            1050            910      1965             0    98136  47.5208   
5            3890           1530      2001             0    98053  47.6561   
8            1050            730      1960             0    98146  47.5123   
10           1860           1700      1965             0    98007  47.6007   
...           ...            ...       ...           ...      ...      ...   
21588        1020            190      2007             0    98117  47.6756   
21590        3110           1800      2007             0    98074  47.6502   
21595        1480             50      2006             0    98103  47.6533   
21606        2600            910      2009             0    98136  47.5537   
21607        1180            130      2008             0    98116  47.5773   

          long  sqft_living15  sqft_lot15  
1     -122.319           1690        7639  
3     -122.393           1360        5000  
5     -122.005           4760      101930  
8     -122.337           1780        8113  
10    -122.145           2210        8925  
...        ...            ...         ...  
21588 -122.375           1210        1118  
21590 -122.066           4560       11063  
21595 -122.346           1530        1282  
21606 -122.398           2050        6200  
21607 -122.409           1330        1265  

[8487 rows x 21 columns]

_Дома с видом на набережную:_

In [67]:
df[df['waterfront'] != 0]

id             date      price  bedrooms  bathrooms   
49      822039084  20150311T000000  1350000.0         3       2.50  \
230    8096000060  20150413T000000   655000.0         2       1.75   
246    2025069065  20140929T000000  2400000.0         4       2.50   
264    2123039032  20141027T000000   369900.0         1       0.75   
300    3225069065  20140624T000000  3075000.0         4       5.00   
...           ...              ...        ...       ...        ...   
19984  2025069140  20150317T000000  1898000.0         3       2.50   
20325   518500480  20140811T000000  3000000.0         3       3.50   
20767  8043700105  20150417T000000  2300000.0         4       4.00   
21201   518500460  20141008T000000  2230000.0         3       3.50   
21576  9253900271  20150107T000000  3567000.0         5       4.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
49            2753     65005     1.0           1     2  ...      9  \
230           1450     15798     2.0           1     4  ...      7   
246           3650      8354     1.0           1     4  ...      9   
264            760     10079     1.0           1     4  ...      5   
300           4550     18641     1.0           1     4  ...     10   
...            ...       ...     ...         ...   ...  ...    ...   
19984         2830      4334     3.0           1     4  ...     10   
20325         4410     10756     2.0           1     4  ...     11   
20767         4360      8175     2.5           1     4  ...     10   
21201         3760      5634     2.0           1     4  ...     11   
21576         4850     10584     2.0           1     4  ...     10   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
49           2165            588      1953             0    98070  47.4041  \
230          1230            220      1915          1978    98166  47.4497   
246          1830           1820      2000             0    98074  47.6338   
264           760              0      1936             0    98070  47.4683   
300          2600           1950      2002             0    98074  47.6053   
...           ...            ...       ...           ...      ...      ...   
19984        2830              0      2006             0    98074  47.6318   
20325        3430            980      2014             0    98056  47.5283   
20767        3940            420      2007             0    98008  47.5724   
21201        2830            930      2014             0    98056  47.5285   
21576        3540           1310      2007             0    98008  47.5943   

          long  sqft_living15  sqft_lot15  
49    -122.451           2680       72513  
230   -122.375           2030       13193  
246   -122.072           3120       18841  
264   -122.438           1230       14267  
300   -122.077           4550       19508  
...        ...            ...         ...  
19984 -122.071           2830       38211  
20325 -122.205           3550        5634  
20767 -122.104           2670        8525  
21201 -122.205           3560        5762  
21576 -122.110           3470       18270  

[163 rows x 21 columns]

In [68]:
f"Количество домов с видом на набережную, как минимум с тремя ванными и с подвалом равно {df[(df['waterfront'] != 0) & (df['bathrooms'] >= 3) & (df['sqft_basement'] > 0)].shape[0]}."

'Количество домов с видом на набережную, как минимум с тремя ванными и с подвалом равно 41.'

_Варианты искомых домов:_

In [69]:
df[(df['waterfront'] != 0) & (df['bathrooms'] >= 3) & (df['sqft_basement'] > 0)]

id             date      price  bedrooms  bathrooms   
300    3225069065  20140624T000000  3075000.0         4       5.00  \
457    2122039094  20141126T000000   705000.0         3       3.00   
1164   1247600105  20141020T000000  5110800.0         5       5.25   
1261   7636800041  20140625T000000   995000.0         3       4.50   
1315   7558700030  20150413T000000  5300000.0         6       6.00   
1361   1925069082  20150511T000000  2200000.0         5       4.25   
2626   7738500731  20140815T000000  4500000.0         5       5.50   
2864   4114601570  20141118T000000  3600000.0         3       3.25   
3914   9808700762  20140611T000000  7062500.0         5       4.50   
3934   1269200229  20140723T000000  1379900.0         3       3.25   
5760    623059016  20140717T000000  1100000.0         4       3.25   
6233   2024069008  20140619T000000  2200000.0         5       4.75   
6508   4217402115  20150421T000000  3650000.0         6       4.75   
7989   9362000040  20140623T000000  3400000.0         3       4.50   
8092   1924059029  20140617T000000  4668000.0         5       6.75   
8163   1724069079  20150319T000000  1452000.0         2       3.25   
8784   3343302110  20150306T000000  1800000.0         3       3.00   
10263   625059036  20140813T000000  2700000.0         5       4.00   
11257  3761100045  20140618T000000  3000000.0         4       4.25   
11402  3426049284  20140819T000000  2300000.0         4       3.25   
11424  5119400075  20140620T000000   950000.0         3       3.25   
11446   624069035  20141209T000000  2750000.0         4       4.00   
11844  2623039082  20150218T000000   770000.0         3       3.50   
12459  4114601580  20140724T000000  1900000.0         6       4.00   
12935  3222049044  20140612T000000   835000.0         3       3.00   
13710  2923500230  20141216T000000  2600000.0         4       4.50   
13967  3126059023  20150303T000000  3395000.0         4       3.50   
14083  3625059043  20140904T000000  3300000.0         5       4.75   
14619  7936500109  20140725T000000  2230000.0         3       3.00   
14822  3623500408  20150330T000000  2600000.0         3       3.00   
15482   624069108  20140812T000000  3200000.0         4       3.25   
15668  4030100290  20141001T000000  1680000.0         5       3.50   
16817   825059178  20140923T000000  2574000.0         4       3.75   
17767  7352200450  20150115T000000  2050000.0         4       3.25   
18199  3625059152  20141230T000000  3300000.0         3       3.25   
18455  8043700300  20140608T000000  2700000.0         4       3.25   
18556  6613000930  20140902T000000  2950000.0         4       3.25   
20325   518500480  20140811T000000  3000000.0         3       3.50   
20767  8043700105  20150417T000000  2300000.0         4       4.00   
21201   518500460  20141008T000000  2230000.0         3       3.50   
21576  9253900271  20150107T000000  3567000.0         5       4.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
300           4550     18641     1.0           1     4  ...     10  \
457           1970     20978     2.0           1     3  ...      9   
1164          8010     45517     2.0           1     4  ...     12   
1261          4380     47044     2.0           1     3  ...      9   
1315          7390     24829     2.0           1     4  ...     12   
1361          4640     22703     2.0           1     4  ...      8   
2626          6640     40014     2.0           1     4  ...     12   
2864          5020     12431     2.0           1     4  ...     10   
3914         10040     37325     2.0           1     2  ...     11   
3934          3786     38038     1.0           1     4  ...      9   
5760          3190     11774     2.0           1     4  ...      8   
6233          5990     10450     2.0           1     4  ...     11   
6508          5480     19401     1.5           1     4  ...     11   
7989          5230     17826     2.0           1     4  ...     10   
8092          9640     1

#### 5.2 Выберите дом клиенту

Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

_Дома с очень красивым видом из окна:_

In [70]:
df[df['view'] == 4]

id             date      price  bedrooms  bathrooms   
21     2524049179  20140826T000000  2000000.0         3       2.75  \
58     7922800400  20140827T000000   951000.0         5       3.25   
230    8096000060  20150413T000000   655000.0         2       1.75   
246    2025069065  20140929T000000  2400000.0         4       2.50   
264    2123039032  20141027T000000   369900.0         1       0.75   
...           ...              ...        ...       ...        ...   
20767  8043700105  20150417T000000  2300000.0         4       4.00   
20923  7852070210  20140527T000000  1149000.0         4       3.00   
21050  2424059170  20150219T000000   900000.0         5       6.00   
21201   518500460  20141008T000000  2230000.0         3       3.50   
21576  9253900271  20150107T000000  3567000.0         5       4.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
21            3050     44867     1.0           0     4  ...      9  \
58            3250     14342     2.0           0     4  ...      8   
230           1450     15798     2.0           1     4  ...      7   
246           3650      8354     1.0           1     4  ...      9   
264            760     10079     1.0           1     4  ...      5   
...            ...       ...     ...         ...   ...  ...    ...   
20767         4360      8175     2.5           1     4  ...     10   
20923         5940     11533     2.0           0     4  ...     11   
21050         7120     40806     2.0           0     4  ...     12   
21201         3760      5634     2.0           1     4  ...     11   
21576         4850     10584     2.0           1     4  ...     10   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
21           2330            720      1968             0    98040  47.5316  \
58           3250              0      1968             0    98008  47.5880   
230          1230            220      1915          1978    98166  47.4497   
246          1830           1820      2000             0    98074  47.6338   
264           760              0      1936             0    98070  47.4683   
...           ...            ...       ...           ...      ...      ...   
20767        3940            420      2007             0    98008  47.5724   
20923        4950            990      2004             0    98065  47.5443   
21050        5480           1640      2007             0    98006  47.5451   
21201        2830            930      2014             0    98056  47.5285   
21576        3540           1310      2007             0    98008  47.5943   

          long  sqft_living15  sqft_lot15  
21    -122.233           4110       20336  
58    -122.116           2960       11044  
230   -122.375           2030       13193  
246   -122.072           3120       18841  
264   -122.438           1230       14267  
...        ...            ...         ...  
20767 -122.104           2670        8525  
20923 -121.870           4240       12813  
21050 -122.114           3440       36859  
21201 -122.205           3560        5762  
21576 -122.110           3470       18270  

[319 rows x 21 columns]

_Дома с с видом на набережную:_

In [71]:
df[df['waterfront'] == 1]

id             date      price  bedrooms  bathrooms   
49      822039084  20150311T000000  1350000.0         3       2.50  \
230    8096000060  20150413T000000   655000.0         2       1.75   
246    2025069065  20140929T000000  2400000.0         4       2.50   
264    2123039032  20141027T000000   369900.0         1       0.75   
300    3225069065  20140624T000000  3075000.0         4       5.00   
...           ...              ...        ...       ...        ...   
19984  2025069140  20150317T000000  1898000.0         3       2.50   
20325   518500480  20140811T000000  3000000.0         3       3.50   
20767  8043700105  20150417T000000  2300000.0         4       4.00   
21201   518500460  20141008T000000  2230000.0         3       3.50   
21576  9253900271  20150107T000000  3567000.0         5       4.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
49            2753     65005     1.0           1     2  ...      9  \
230           1450     15798     2.0           1     4  ...      7   
246           3650      8354     1.0           1     4  ...      9   
264            760     10079     1.0           1     4  ...      5   
300           4550     18641     1.0           1     4  ...     10   
...            ...       ...     ...         ...   ...  ...    ...   
19984         2830      4334     3.0           1     4  ...     10   
20325         4410     10756     2.0           1     4  ...     11   
20767         4360      8175     2.5           1     4  ...     10   
21201         3760      5634     2.0           1     4  ...     11   
21576         4850     10584     2.0           1     4  ...     10   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
49           2165            588      1953             0    98070  47.4041  \
230          1230            220      1915          1978    98166  47.4497   
246          1830           1820      2000             0    98074  47.6338   
264           760              0      1936             0    98070  47.4683   
300          2600           1950      2002             0    98074  47.6053   
...           ...            ...       ...           ...      ...      ...   
19984        2830              0      2006             0    98074  47.6318   
20325        3430            980      2014             0    98056  47.5283   
20767        3940            420      2007             0    98008  47.5724   
21201        2830            930      2014             0    98056  47.5285   
21576        3540           1310      2007             0    98008  47.5943   

          long  sqft_living15  sqft_lot15  
49    -122.451           2680       72513  
230   -122.375           2030       13193  
246   -122.072           3120       18841  
264   -122.438           1230       14267  
300   -122.077           4550       19508  
...        ...            ...         ...  
19984 -122.071           2830       38211  
20325 -122.205           3550        5634  
20767 -122.104           2670        8525  
21201 -122.205           3560        5762  
21576 -122.110           3470       18270  

[163 rows x 21 columns]

_Дома в очень хорошем состоянии:_

In [72]:
df[df['condition'] == 5]

id             date      price  bedrooms  bathrooms   
3      2487200875  20141209T000000   604000.0         4       3.00  \
25     1202000200  20141103T000000   233000.0         3       2.00   
27     3303700376  20141201T000000   667000.0         3       1.00   
49      822039084  20150311T000000  1350000.0         3       2.50   
62     5072410070  20141021T000000   505000.0         3       1.75   
...           ...              ...        ...       ...        ...   
19604  7518504130  20140626T000000   663000.0         3       2.00   
19620  3902100175  20140728T000000   850000.0         5       3.00   
19873  3449000060  20141001T000000   320000.0         3       1.00   
19905  2888000030  20140926T000000   500000.0         4       2.25   
21303   522059130  20150429T000000   465000.0         3       1.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
3             1960      5000     1.0           0     0  ...      7  \
25            1710      4697     1.5           0     0  ...      6   
27            1400      1581     1.5           0     0  ...      8   
49            2753     65005     1.0           1     2  ...      9   
62            2519      8690     2.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
19604         1480      3876     1.0           0     0  ...      7   
19620         3900      5250     1.5           0     1  ...      8   
19873         1400      9000     1.0           0     0  ...      7   
19905         2270      8196     1.0           0     0  ...      7   
21303         1150     18200     1.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
3            1050            910      1965             0    98136  47.5208  \
25           1710              0      1941             0    98002  47.3048   
27           1400              0      1909             0    98112  47.6221   
49           2165            588      1953             0    98070  47.4041   
62           2519              0      1973             0    98166  47.4428   
...           ...            ...       ...           ...      ...      ...   
19604         860            620      1928             0    98117  47.6808   
19620        2620           1280      1931             0    98116  47.5577   
19873        1400              0      1959             0    98059  47.5022   
19905        1150           1120      1963             0    98034  47.7214   
21303        1150              0      1959             0    98058  47.4262   

          long  sqft_living15  sqft_lot15  
3     -122.393           1360        5000  
25    -122.218           1030        4705  
27    -122.314           1860        3861  
49    -122.451           2680       72513  
62    -122.344           2500        9500  
...        ...            ...         ...  
19604 -122.382           1660        3774  
19620 -122.389           1950        5700  
19873 -122.145           1440        8400  
19905 -122.227           1920       10122  
21303 -122.187           1714       18200  

[1701 rows x 21 columns]

_Дома, год постройки которых не ранее 1980 года:_

In [73]:
df[df['yr_built'] >= 1980]

id             date      price  bedrooms  bathrooms   
4      1954400510  20150218T000000   510000.0         3       2.00  \
5      7237550310  20140512T000000  1225000.0         4       4.50   
6      1321400060  20140627T000000   257500.0         3       2.25   
9      3793500160  20150312T000000   323000.0         3       2.50   
16     1875500060  20140731T000000   395000.0         3       2.00   
...           ...              ...        ...       ...        ...   
21608   263000018  20140521T000000   360000.0         3       2.50   
21609  6600060120  20150223T000000   400000.0         4       2.50   
21610  1523300141  20140623T000000   402101.0         2       0.75   
21611   291310100  20150116T000000   400000.0         3       2.50   
21612  1523300157  20141015T000000   325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
4             1680      8080     1.0           0     0  ...      8  \
5             5420    101930     1.0           0     0  ...     11   
6             1715      6819     2.0           0     0  ...      7   
9             1890      6560     2.0           0     0  ...      7   
16            1890     14040     2.0           0     0  ...      7   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
4            1680              0      1987             0    98074  47.6168  \
5            3890           1530      2001             0    98053  47.6561   
6            1715              0      1995             0    98003  47.3097   
9            1890              0      2003             0    98038  47.3684   
16           1890              0      1994             0    98019  47.7277   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
4     -122.045           1800        7503  
5     -122.005           4760      101930  
6     -122.327           2238        6819  
9     -122.031           2390        7570  
16    -121.962           1890       14018  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[9279 rows x 21 columns]

 _Варианты искомых домов:_

In [74]:
df[((df['view'] == 4) | (df['waterfront'] == 1)) & (df['condition'] == 5) & (df['yr_built'] >= 1980)]

id             date      price  bedrooms  bathrooms   
11257  3761100045  20140618T000000  3000000.0         4       4.25  \
14682  7278100515  20140821T000000  1295000.0         2       2.50   
17312  3585901025  20140613T000000  1735000.0         3       2.50   
18991  7856400300  20140702T000000  1411600.0         2       2.50   
18992  7856400300  20150322T000000  1505000.0         2       2.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
11257         4850     12445     2.0           1     4  ...     10  \
14682         2910     19449     2.0           1     4  ...      9   
17312         4310     32093     1.5           0     4  ...     10   
18991         3180      9400     2.0           0     4  ...     10   
18992         3180      9400     2.0           0     4  ...     10   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
11257        3850           1000      1989             0    98034  47.7011  \
14682        1940            970      1985             0    98177  47.7729   
17312        2510           1800      1982             0    98177  47.7624   
18991        2610            570      1985             0    98006  47.5617   
18992        2610            570      1985             0    98006  47.5617   

          long  sqft_living15  sqft_lot15  
11257 -122.244           3350       12210  
14682 -122.393           2540       23598  
17312 -122.380           3810       26400  
18991 -122.158           3760        9450  
18992 -122.158           3760        9450  

[5 rows x 21 columns]

_Цены искомых домов:_

In [75]:
df[((df['view'] == 4) | (df['waterfront'] == 1)) & (df['condition'] == 5) & (df['yr_built'] >= 1980)]['price']

11257    3000000.0
14682    1295000.0
17312    1735000.0
18991    1411600.0
18992    1505000.0
Name: price, dtype: float64

_Ценовой диапазон искомых домов:_

In [76]:
looking_for_houses = df[((df['view'] == 4) | (df['waterfront'] == 1)) & (df['condition'] == 5) & (df['yr_built'] >= 1980)]['price']
looking_for_houses.min(), looking_for_houses.max()

(1295000.0, 3000000.0)

#### 5.3 Выберите дом клиенту

Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

_Дома без подвала:_

In [84]:
df[df['sqft_basement'] == 0]

id             date     price  bedrooms  bathrooms   
0      7129300520  20141013T000000  221900.0         3       1.00  \
2      5631500400  20150225T000000  180000.0         2       1.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
6      1321400060  20140627T000000  257500.0         3       2.25   
7      2008000270  20150115T000000  291850.0         3       1.50   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
0             1180      5650     1.0           0     0  ...      7  \
2              770     10000     1.0           0     0  ...      6   
4             1680      8080     1.0           0     0  ...      8   
6             1715      6819     2.0           0     0  ...      7   
7             1060      9711     1.0           0     0  ...      7   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
0            1180              0      1955             0    98178  47.5112  \
2             770              0      1933             0    98028  47.7379   
4            1680              0      1987             0    98074  47.6168   
6            1715              0      1995             0    98003  47.3097   
7            1060              0      1963             0    98198  47.4095   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
2     -122.233           2720        8062  
4     -122.045           1800        7503  
6     -122.327           2238        6819  
7     -122.315           1650        9711  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[13126 rows x 21 columns]

_Доса с двумя этажами:_

In [79]:
df[df['floors'] == 2]

id             date     price  bedrooms  bathrooms   
1      6414100192  20141209T000000  538000.0         3       2.25  \
6      1321400060  20140627T000000  257500.0         3       2.25   
9      3793500160  20150312T000000  323000.0         3       2.50   
15     9297300055  20150124T000000  650000.0         4       3.00   
16     1875500060  20140731T000000  395000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21607  2997800021  20150219T000000  475000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
1             2570      7242     2.0           0     0  ...      7  \
6             1715      6819     2.0           0     0  ...      7   
9             1890      6560     2.0           0     0  ...      7   
15            2950      5000     2.0           0     3  ...      9   
16            1890     14040     2.0           0     0  ...      7   
...            ...       ...     ...         ...   ...  ...    ...   
21607         1310      1294     2.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
1            2170            400      1951          1991    98125  47.7210  \
6            1715              0      1995             0    98003  47.3097   
9            1890              0      2003             0    98038  47.3684   
15           1980            970      1979             0    98126  47.5714   
16           1890              0      1994             0    98019  47.7277   
...           ...            ...       ...           ...      ...      ...   
21607        1180            130      2008             0    98116  47.5773   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
1     -122.319           1690        7639  
6     -122.327           2238        6819  
9     -122.031           2390        7570  
15    -122.375           2140        4000  
16    -121.962           1890       14018  
...        ...            ...         ...  
21607 -122.409           1330        1265  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[8241 rows x 21 columns]

_Дома стоимостью до 150000:_

In [81]:
df[df['price'] < 150000]

id             date     price  bedrooms  bathrooms   
465    8658300340  20140523T000000   80000.0         1       0.75  \
704      41000454  20140815T000000  130000.0         2       1.00   
876    8802400416  20150213T000000  147500.0         3       1.00   
1066   3340401535  20141105T000000  140000.0         1       1.00   
1149   3421079032  20150217T000000   75000.0         1       0.00   
...           ...              ...       ...       ...        ...   
19249  4337600205  20141112T000000  129888.0         2       1.00   
19384  9320901250  20140910T000000  133400.0         3       1.00   
19452  3980300371  20140926T000000  142000.0         0       0.00   
19524  3306200230  20150303T000000  147000.0         3       1.50   
20962  1278000210  20150311T000000  110000.0         2       1.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
465            430      5050     1.0           0     0  ...      4  \
704            880      9000     1.0           0     0  ...      5   
876           1530      8498     1.0           0     0  ...      7   
1066           730      6890     1.0           0     0  ...      4   
1149           670     43377     1.0           0     0  ...      3   
...            ...       ...     ...         ...   ...  ...    ...   
19249          710      9900     1.0           0     0  ...      6   
19384          900      2550     1.0           0     0  ...      6   
19452          290     20875     1.0           0     0  ...      1   
19524         1480      9606     1.0           0     0  ...      7   
20962          828      4524     1.0           0     0  ...      6   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
465           430              0      1912             0    98014  47.6499  \
704           880              0      1928             0    98188  47.4672   
876          1530              0      1959             0    98031  47.4040   
1066          730              0      1926             0    98055  47.4670   
1149          670              0      1966             0    98022  47.2638   
...           ...            ...       ...           ...      ...      ...   
19249         710              0      1943             0    98166  47.4790   
19384         900              0      1978             0    98023  47.3036   
19452         290              0      1963             0    98024  47.5308   
19524        1100            380      1964             0    98023  47.2978   
20962         828              0      1968          2007    98001  47.2655   

          long  sqft_living15  sqft_lot15  
465   -121.909           1200        7500  
704   -122.291           1410       10000  
876   -122.203           1380        8498  
1066  -122.215           1790        7969  
1149  -121.906           1160       42882  
...        ...            ...         ...  
19249 -122.339           1070        9900  
19384 -122.363           1120        2550  
19452 -121.888           1620       22850  
19524 -122.363           1600        9619  
20962 -122.244            828        5402  

[173 rows x 21 columns]

Варианты искомых домов:

In [85]:
df[(df['sqft_basement'] == 0) & (df['floors'] == 2) & (df['price'] < 150000)]

id             date     price  bedrooms  bathrooms   
2589   5061300030  20140508T000000  134000.0         2       1.50  \
10105  5466310060  20150324T000000  139500.0         2       1.50   
15610  7614100080  20150211T000000  140000.0         3       1.75   
18332  9272202260  20140924T000000  130000.0         3       1.00   
18939  4239400300  20141129T000000   90000.0         3       1.00   
18988  7399100210  20141126T000000  140000.0         3       1.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade   
2589           980      5000     2.0           0     0  ...      7  \
10105         1230      1561     2.0           0     0  ...      7   
15610         1270      8991     2.0           0     0  ...      7   
18332         1200      7000     2.0           0     0  ...      7   
18939          980      2490     2.0           0     0  ...      6   
18988         1200      2002     2.0           0     0  ...      8   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat   
2589          980              0      1922          2003    98014  47.7076  \
10105        1230              0      1983             0    98042  47.3768   
15610        1270              0      1981             0    98042  47.3563   
18332        1200              0      1908             0    98116  47.5883   
18939         980              0      1969             0    98092  47.3170   
18988        1200              0      1966             0    98055  47.4659   

          long  sqft_living15  sqft_lot15  
2589  -121.359           1040        5000  
10105 -122.149           1660        2243  
15610 -122.149           1270        8993  
18332 -122.384           3290        6000  
18939 -122.182            980        3154  
18988 -122.189           1270        1848  

[6 rows x 21 columns]

In [91]:
f"Оценка состояния искомых домов в среднем равна {round(df[(df['sqft_basement'] == 0) & (df['floors'] == 2) & (df['price'] < 150000)]['condition'].mean(), 3)}."

'Оценка состояния искомых домов в среднем равна 2.833.'